pkg/handlers/v1/filter.go

```diff
@@ -0,0 +1,72 @@
+package v1
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/asecurityteam/awsconfig-filterd/pkg/domain"
+	"github.com/asecurityteam/awsconfig-filterd/pkg/logs"
+)
+
+// ConfigNotification is the basic structure of all incoming AWS Config SNS events
+// (https://docs.aws.amazon.com/config/latest/developerguide/example-sns-notification.html).
+type ConfigNotification struct {
+	Message            string `json:"Message"`
+	Timestamp          string `json:"Timestamp"`
+	ProcessedTimestamp string `json:"ProcessedTimestamp"`
+	Type               string `json:"Type"`
+	MessageID          string `json:"MessageId"`
+	TopicArn           string `json:"TopicArn"`
+	Subject            string `json:"Subject"`
+	SignatureVersion   string `json:"SignatureVersion"`
+	Signature          string `json:"Signature"`
+	SigningCertURL     string `json:"SigningCertURL"`
+	UnsubscribeURL     string `json:"UnsubscribeURL"`
+}
+
+// ConfigEvent represents a single AWS Config Configuration Item.
+type ConfigEvent struct {
+	ConfigurationItem domain.ConfigurationItem `json:"configurationItem"`
+}
+
+// ConfigFilterHandler applies a filter to AWS Config events.
+type ConfigFilterHandler struct {
+	Filter domain.ConfigFilter
+	LogFn  domain.LogFn
+	StatFn domain.StatFn
+}
+
+// Handle accepts Config events, applies a filter, and returns the events that match.
+func (h *ConfigFilterHandler) Handle(ctx context.Context, in ConfigNotification) (ConfigNotification, error) {
+	logger := h.LogFn(ctx)
+	stater := h.StatFn(ctx)
+
+	var event ConfigEvent
+	e := json.Unmarshal([]byte(in.Message), &event)
+	if e != nil {
+		logger.Error(logs.InvalidInput{Reason: e.Error()})
+		return ConfigNotification{}, domain.ErrEventDiscarded{Reason: e.Error()}
+	}
+
+	if event.ConfigurationItem.ResourceType == "" {
+		e = domain.ErrEventDiscarded{Reason: "empty resource type"}
+		logger.Info(logs.InvalidInput{Reason: e.Error()})
```

Comment:
I am on the fence about this, so I'd like to hear some input.

I think a case there there is no resource type provided would be an anomaly that we'd want to know about. Receiving not resource type would mean something is wrong with the AWS config output. Since our input is not generated by a user, I think logging at an error level here wouldn't be a bad idea. What are your thoughts?

Reply:
That reasoning makes sense to me, I'll bump this to the error level.